File: XJYChartDemo/XJYChartDemo/View/AreaLineTableViewCell.m
//
//  AreaLineTableViewCell.m
//  RecordLife
//
//  Created by 谢俊逸 on 09/05/2017.
//  Copyright © 2017 谢俊逸. All rights reserved.
//

#import "AreaLineTableViewCell.h"

#import <XJYChart/XChart.h>
@implementation AreaLineTableViewCell

- (instancetype)initWithStyle:(UITableViewCellStyle)style
              reuseIdentifier:(NSString*)reuseIdentifier {
  if (self = [super initWithStyle:style reuseIdentifier:reuseIdentifier]) {
    NSMutableArray* itemArray = [[NSMutableArray alloc] init];

    NSMutableArray* numberArray = [NSMutableArray new];

    numberArray =
        [NSMutableArray arrayWithArray:@[ @75, @63, @183, @109, @88 ]];

    XLineChartItem* item =
        [[XLineChartItem alloc] initWithDataNumberArray:numberArray
                                                  color:XJYWhite];
    [itemArray addObject:item];

    XAreaLineChartConfiguration* configuration =
        [[XAreaLineChartConfiguration alloc] init];
    configuration.isShowPoint = YES;
    configuration.lineMode = CurveLine;
    XLineChart* lineChart =
        [[XLineChart alloc] initWithFrame:CGRectMake(0, 0, 375, 200)
                            dataItemArray:itemArray
                        dataDiscribeArray:[NSMutableArray arrayWithArray:@[
                          @"January", @"February", @"March", @"April", @"May"
                        ]]
                                topNumber:@240
                             bottomNumber:@0
                                graphMode:AreaLineGraph
                       chartConfiguration:configuration];

    [self.contentView addSubview:lineChart];
    self.selectionStyle = UITableViewCellSelectionStyleNone;
  }
  return self;
}

- (void)awakeFromNib {
  [super awakeFromNib];
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
  [super setSelected:selected animated:animated];
}

@end

File: XJYChartDemo/XJYChartDemo-Swift/View/AreaLineTableViewCell.swift
//
//  AreaLineTableViewCell.swift
//  XJYChartDemo-Swift
//
//  Created by Kelly Roach on 8/17/18.
//  Copyright © 2018 JunyiXie. All rights reserved.
//

import UIKit
import XJYChart

class AreaLineTableViewCell : UITableViewCell {
    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)

        var itemArray: [AnyHashable] = []
        let numberArray = [75, 63, 183, 109, 88]
        let item = XLineChartItem(dataNumber: NSMutableArray(array:numberArray), color: Constants.XJYWhite)
        itemArray.append(item!)
        let configuration = XAreaLineChartConfiguration()
        configuration.isShowPoint = true
        configuration.lineMode = XLineMode.CurveLine
        configuration.ordinateDenominator = 6
            //    configuration.isEnableNumberLabel = YES;
        let lineChart = XLineChart(frame: CGRect(x: 0, y: 0, width: 375, height: 200), dataItemArray: NSMutableArray(array: itemArray), dataDiscribeArray: ["January", "February", "March", "April", "May"], topNumber: 240, bottomNumber: 0, graphMode: XLineGraphMode.AreaLineGraph, chartConfiguration: nil)
        contentView.addSubview(lineChart!)
        selectionStyle = UITableViewCellSelectionStyle.none
    
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)
    }
}
